merkle-tree/indexed/tests/tests.rs

```diff
@@ -540,113 +540,115 @@ pub fn functional_non_inclusion_test() {
-        .unwrap();
-    let address_bundle_1 = relayer_indexing_array
-        .new_element_with_low_element_index(old_low_address_1.index, &address_1)
-        .unwrap();
-    let change_log_index = onchain_indexed_merkle_tree.changelog_index();
-    {
-        let mut low_element_proof_1 = relayer_merkle_tree
-            .get_proof_of_leaf(old_low_address_1.index, false)
-            .unwrap();
-
-        onchain_indexed_merkle_tree
-            .update(
-                change_log_index.clone(),
-                address_bundle_1.new_element.clone(),
-                old_low_address_1.clone(),
-                old_low_address_next_value_1,
-                &mut low_element_proof_1,
-            )
-            .unwrap();
-    }
-
-    // getting parameters for the second actor with the pre update state
-    let (old_low_address, old_low_address_next_value) = actor_2_indexed_array_state
-        .find_low_element(&address_1)
-        .unwrap();
-    let address_bundle = relayer_indexing_array
-        .new_element_with_low_element_index(old_low_address.index, &address_2)
-        .unwrap();
-
-    let mut low_element_proof = relayer_merkle_tree
-        .get_proof_of_leaf(old_low_address.index, false)
-        .unwrap();
-
-    onchain_indexed_merkle_tree
-        .update(
-            change_log_index,
-            address_bundle.new_element,
-            old_low_address,
-            old_low_address_next_value,
-            &mut low_element_proof,
-        )
-        .unwrap();
-
-    // update the relayer state
-    relayer_merkle_tree
-        .append(&address_1, &mut relayer_indexing_array)
-        .unwrap();
-
-    relayer_merkle_tree
-        .append(&address_2, &mut relayer_indexing_array)
-        .unwrap();
-}
+// /**
+//  *
+//  * Range Hash (value, next_index, next_value) -> need next value not next value index
+//  * Update of a range:
+//  * 1. Find the low element, low element points to the next hight element
+//  * 2. update low element with H (low_value, new_inserted_value_index, new_inserted_value)
+//  * 3. append the tree with H(new_inserted_value,index_of_next_value, next_value)
+//  *
+//  */
+// /// This test is generating a situation where the low element has to be patched.
+// /// Scenario:
+// /// 1. two parties start with the initialized indexing array
+// /// 2. both parties compute their values with the empty indexed Merkle tree state
+// /// 3. party one inserts first
+// /// 4. party two needs to patch the low element because the low element has changed
+// /// 5. party two inserts
+// Commented because the test is not working
+// TODO: figure out address Merkle tree changelog
+// #[test]
+// pub fn functional_changelog_test() {
+//     let address_1 = 30_u32.to_biguint().unwrap();
+//     let address_2 = 10_u32.to_biguint().unwrap();
+
+//     perform_change_log_test(address_1.clone(), address_2.clone());
+// }
+
+// fn perform_change_log_test(address_1: BigUint, address_2: BigUint) {
```

Comment:
Is this test commented because of the broken concurrency problems you talked about? I'm happy to help tomorrow with debugging it, if it's still a blocker.

Reply:
This code is commented because the concurrency feature was breaking my test.
It's not directly a blocker my tests run since I removed concurrency.
But we don't have address Merkle tree concurrency.
We definitely need address Merkle tree concurrency for mainnet.
But for this audit I want to get everything else working before looking into concurrency again.